### PR TITLE
refactor!: VCs and proper DID documents as basic identity modules in SDK 

### PR DESCRIPTION
**Pull request by rflechtner**
## fixes KILTProtocol/ticket#2156

This PR is the combined result of refactoring efforts over the last couple of months, making the following improvements:

- The default representation of Kilt Credentials has been transitioned to one conforming to the VC Data Model 1.0.
  - Functions to support and interact with the previous representation is now available from the new package `@kiltprotocol/legacy-credentials`.
  - The new functions are available from the new package `@kiltprotocol/credentials`.
- The Did resolver is now fully specification-conformant, implementing `resolve`, `resolveRepresentation`, and `dereference` as defined in https://www.w3.org/TR/did-core/#resolution.
- Implements Verifiable Presentations and VC Data Integrity 1.0 for presentation signing
  - We implemented 3 cryptosuites for this purpose, compatible with Digital Bazaar’s implementation
- Provides high-level implementations of common use cases, e.g.:
  - Resolving a DID
  - Issuing a VC
  - Creating a Presentation from VCs
  - Verifying a VC or a VP, checking the revocation status of a VC
- Removal of low-level functions from `sdk-js`.
  - These are still available to be imported directly from the packages in which they are defined.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)
    * Either PR or Ticket to update [the Docs](https://github.com/KILTprotocol/docs)
    * Link the PR/Ticket here

### CODE REVIEW

File: packages/asset-credentials/src/credentials/PublicCredential.spec.ts [hunk 1]
```diff
@@ -8,11 +8,11 @@
...
 
 import { ConfigService } from '@kiltprotocol/config'
-import { CType } from '@kiltprotocol/core'
+import { CType } from '@kiltprotocol/credentials'
 import * as Did from '@kiltprotocol/did'
 import type {
-  AssetDidUri,
-  DidUri,
+  AssetDid,
+  Did as KiltDid,
```
Comment (Dudleyneedham) [inline packages/asset-credentials/src/credentials/PublicCredential.spec.ts:15]: Can you Declare this where it is used?

File: packages/asset-credentials/src/credentials/PublicCredential.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ import type { AccountId } from '@polkadot/types/interfaces'
 import type { PublicCredentialsCredentialsCredential } from '@kiltprotocol/augment-api'
 import type {
   HexString,
-  DidUri,
+  Did as KiltDid,
```
Comment (Dudleyneedham) [inline packages/asset-credentials/src/credentials/PublicCredential.ts:12]: Same as above.

File: packages/credentials/src/V1/KiltAttestationProofV1.spec.ts [hunk 3]
```diff
@@ -8,30 +8,29 @@
...
-import { Credential } from '@kiltprotocol/core'
 import { parse } from '@kiltprotocol/did'
-import type { DidUri } from '@kiltprotocol/types'
+import type { Did } from '@kiltprotocol/types'
 
 import {
   attestation,
   blockHash,
-  credential,
+  credential as VC,
```
Comment (Dudleyneedham) [inline packages/credentials/src/V1/KiltAttestationProofV1.spec.ts:17]: Please can we do this where we are using it?

File: packages/credentials/src/V1/KiltCredentialV1.spec.ts [hunk 4]
```diff
@@ -5,34 +5,18 @@
...
 
-import { CType } from '@kiltprotocol/core'
-import { randomAsHex, randomAsU8a } from '@polkadot/util-crypto'
+import { randomAsHex } from '@polkadot/util-crypto'
+
+import * as CType from '../ctype/index.js'
 import {
-  attestation,
-  credential,
+  credential as VC,
```
Comment (Dudleyneedham) [inline packages/credentials/src/V1/KiltCredentialV1.spec.ts:12]: NAUGHTY

File: packages/credentials/src/attestation/Attestation.ts [hunk 5]
```diff
@@ -9,28 +9,27 @@ import type {
...
+  Did as KiltDid,
 } from '@kiltprotocol/types'
 import { DataUtils, SDKErrors } from '@kiltprotocol/utils'
 import * as Did from '@kiltprotocol/did'
 import { DelegationNode } from '../delegation/DelegationNode.js'
-import * as Credential from '../credential/index.js'
 
 /**
- * An [[Attestation]] certifies a [[Claim]], sent by a claimer in the form of a [[Credential]]. [[Attestation]]s are **written on the blockchain** and are **revocable**.
+ * An {@link IAttestation | Attestation} certifies a {@link IClaim | Claim}, sent by a claimer in the form of a {@link ICredential | Credential}. Attestations are **written on the blockchain** and are **revocable**.
```
Comment (Dudleyneedham) [inline packages/credentials/src/attestation/Attestation.ts:19]:
```suggestion
 * An {@link IAttestation | Attestation} certifies a {@link IClaim | Claim}, sent by a claimer in the form of a {@link ICredential | Credential}.
Attestations are **written on the blockchain** and are **revocable**.
```

File: packages/did/src/Did.signature.ts [hunk 6]
```diff
@@ -95,22 +99,64 @@ export async function verifyDidSignature({
...
+ * Type guard assuring that the input is a valid DidSignature object, consisting of a signature as hex and the DID URL of the signer's verification method.
  * Does not cryptographically verify the signature itself!
  *
  * @param input Arbitrary input.
  * @returns True if validation of form has passed.
  */
 export function isDidSignature(
   input: unknown
-): input is DidSignature | OldDidSignature {
+): input is DidSignature | OldDidSignatureV1 {
```
Comment (Dudleyneedham) [inline packages/did/src/Did.signature.ts:159]: Why is this called old? SHould this be legacy?

File: packages/utils/src/Signers.ts [hunk 7]
```diff
@@ -0,0 +1,454 @@
...
+  createSigner as es256kSigner,
+  cryptosuite as es256kSuite,
+} from '@kiltprotocol/es256k-jcs-2023'
+import {
+  createSigner as ed25519Signer,
+  cryptosuite as ed25519Suite,
+} from '@kiltprotocol/eddsa-jcs-2022'
+import {
+  cryptosuite as sr25519Suite,
+  createSigner as sr25519Signer,
```
Comment (Dudleyneedham) [inline packages/utils/src/Signers.ts:34]: Can we declare this where we use it

File: package.json [hunk 8]
```diff
@@ -8,13 +8,13 @@
   "license": "BSD-4-Clause",
   "scripts": {
     "check": "tsc -p tsconfig.json --noEmit",
-    "build": "yarn workspaces foreach -p -t --exclude '{root-workspace}' run build",
+    "build": "yarn workspaces foreach -ptAv --exclude '{root-workspace}' run build",
```
[the remark this answers is not in the document]
Comment (rflechtner) [reply package.json:11]: p: parallel
t: topological
A: all packages
v: verbose (more info)

File: packages/utils/src/Signers.ts (hunk 7 above)
Comment (rflechtner) [reply packages/utils/src/Signers.ts:34]: I wanted to make these available, but obviously they need to be identifiable by the algorithm they use. What are you suggesting?

Comment (rflechtner) [reply packages/utils/src/Signers.ts:34]: I suggest if you have a better idea (like exporting both under cryptosuite and createSigner on an object, e.g., `ed25519Suite`) we make a ticket and implement it on a feature branch forked from develop

**Pull request merged**